src/user/user.controller.ts

```diff
@@ -11,21 +12,58 @@ export class UserController {
 
     }
 
+    @Post('login')
+    @UsePipes (new ValidationPipe())
+    login (@Body() data: UserCredentialsDTO){
+        return this.userService.login(data);
+    }
+
+    @Post('changePassword')
+    @UsePipes (new ValidationPipe())
+    @UseGuards(new AuthGuard()) 
+    changePassword (@User('id') userId: string, @Body() data) {
```

Comment:
Typa bien el data para evitarte problemas con la deconstrucción. o en cuyo defecto asegurate que existen esos campos previamente. 
if (!data) throw new HttpException

Si usamos typescript 3.7.4 podemos hacer data?.oldpassword también... vamos usar el operador ? 

Y por favor... usa camelCase para los nombres de las keys "oldpassword" to "oldPassword" etc..